pkg/ring/kv/dynamodb/dynamodb.go

```diff
@@ -161,6 +162,55 @@ func (kv dynamodbKV) Delete(ctx context.Context, key dynamodbKey) error {
 }
 
 func (kv dynamodbKV) Put(ctx context.Context, key dynamodbKey, data []byte) error {
+	input := &dynamodb.PutItemInput{
+		TableName: kv.tableName,
+		Item:      kv.generatePutItemRequest(key, data),
+	}
+	_, err := kv.ddbClient.PutItemWithContext(ctx, input)
+	return err
+}
+
+func (kv dynamodbKV) Batch(ctx context.Context, put map[dynamodbKey][]byte, delete []dynamodbKey) error {
+	var writeRequests []*dynamodb.WriteRequest
+	for key, data := range put {
+		item := kv.generatePutItemRequest(key, data)
+		writeRequests = append(writeRequests, &dynamodb.WriteRequest{
+			PutRequest: &dynamodb.PutRequest{
+				Item: item,
+			},
+		})
+	}
+
+	for _, key := range delete {
+		item := generateItemKey(key)
+
+		writeRequests = append(writeRequests, &dynamodb.WriteRequest{
+			DeleteRequest: &dynamodb.DeleteRequest{
+				Key: item,
+			},
+		})
+	}
+
+	if len(writeRequests) == 0 {
+		return nil
+	}
+
+	input := &dynamodb.BatchWriteItemInput{
+		RequestItems: map[string][]*dynamodb.WriteRequest{
+			*kv.tableName: writeRequests,
+		},
+	}
+
+	resp, err := kv.ddbClient.BatchWriteItemWithContext(ctx, input)
+
+	if resp.UnprocessedItems != nil && len(resp.UnprocessedItems) > 0 {
+		return fmt.Errorf("error processing batch request for %s requests", resp.UnprocessedItems)
```

Comment:
If we have un processed items, what will be the value of `err`? I am wondering should we add `err` in this returned error, otherwise `err` content is skipped in this case

Reply:
true, i have added this if and forgot about err.
added a test for it also.